chpl/chpl-service/src/main/java/gov/healthit/chpl/util/ErrorMessageUtil.java

```diff
@@ -12,8 +12,15 @@
 @Component
 public class ErrorMessageUtil {
 
-    @Autowired private MessageSource messageSource;
-
+    private MessageSource messageSource;
```

Comment:
Is there an advantage to using the autowired constructor vs just having autowired on the variable? Most of the code has it the latter way, just wondering why change it.

Reply:
https://www.vojtechruzicka.com/field-dependency-injection-considered-harmful/

Reply:
I'm ok with this then.. if we want to make this part of our coding style and change classes as we touch them going forward that is fine. Can you mention it at the Thursday meeting so that we are all on the same page? If it's already been mentioned I've obviously forgotten or missed it so please repeat. :)

Reply:
If we're going to move towards constructor based initialization, then doesn't the constructor without the dependency need to be removed? Otherwise we lose out on the biggest advantage of constructor based vs. setter based initialization, in that a "new ErrorMessageUtil()" will have an NPE as soon as someone tries to "getMessage()"

Reply:
Well you should never call new ErrorMessageUtil() if you want to use it a spring component. It should always be autowired into a class in some way otherwise you lose all the springyness of it. Not sure if the default constructor has to be removed but I could get on board with that if it works.

Reply:
Everything I read said one of the reasons you do this with a constructor is so that, when someone instantiates the class not in a spring-y way, they'll be required to pass in the dependency. I think that means we'd need to remove the "non-dependency injected" constructors. As long as it compiles and works without that default one I think we should do that, if we're going down this path

Reply:
@tmy1313  did you undo the changes you made here? I think we got to the point where moving to constructor-injection sounded like the right idea

Reply:
I'm good with this one. @kekey1 are you satisfied with these changes wrt this conversation?

Reply:
Yep!